vault/quotas/quotas_rate_limit_test.go

```diff
@@ -147,46 +147,45 @@ func TestRateLimitQuota_Allow_WithBlock(t *testing.T) {
 
 	var wg sync.WaitGroup
 
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
 	reqFunc := func(addr string, atomicNumAllow, atomicNumFail *atomic.Int32) {
 		defer wg.Done()
 
-		resp, err := rlq.allow(context.Background(), &Request{ClientAddress: addr})
-		if err != nil {
-			return
-		}
+		for ctx.Err() == nil {
```

Comment:
Shouldn't this have another condition based on time/number of loops? If I understand this code correctly, if `rlq.allow` never returns a failure, the test will never timeout.

Reply:
The timeout is embedded in the ctx.

Reply:
Oops completely missed that.